### PR TITLE
use dimension id and href from recipe codelists instead of cantabular response

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

- Populate `id` and `href` dimension values from recipe code lists (if provided), instead of Cantabular response.
- Modified unit tests to validate the dimensions that we send to dataset-api, and validate that the expected values are sent.

trello card: https://trello.com/c/fEsZyzBI/5402-dataset-api-returning-incorrect-dimension-information-for-cantabular-dataset-5

### How to review

- Make sure code changes make sense
- Make sure unit tests pass

### Who can review

Anyone

### CODE REVIEW

File: handler/instance_started.go [hunk 1]
```diff
@@ -197,9 +204,21 @@ func (h *InstanceStarted) createUpdateInstanceRequest(cb cantabular.Codebook, e
 			}
 		}
 
+		id := sourceName
+		url := fmt.Sprintf("%s/code-lists/%s", h.cfg.RecipeAPIURL, sourceName)
+
+		// id and url values overwritten by codelist values.
+		// Note that cantabular codebook is sorted with exactly the same order as codelists array.
```
Comment (DavidSubiros) [inline handler/instance_started.go:211]: This is true for my local testing. Do not merge this PR until this assumption is validated with Fran or some Cantabular documentation?

File: handler/instance_started.go [hunk 2]
```diff
@@ -197,9 +204,21 @@ func (h *InstanceStarted) createUpdateInstanceRequest(cb cantabular.Codebook, e
...
+		if len(codelists[i].ID) > 0 {
```
Comment (redhug1) [inline handler/instance_started.go:212]: would it be more efficient to have the line as:
if codelists[i].ID != "" {

File: handler/instance_started.go [hunk 3]
```diff
@@ -197,9 +204,21 @@ func (h *InstanceStarted) createUpdateInstanceRequest(cb cantabular.Codebook, e
...
+			id = codelists[i].ID
+		}
+		if len(codelists[i].HRef) > 0 {
```
Comment (redhug1) [inline handler/instance_started.go:215]: would it be more efficient to have the line as:
if codelists[i].ID != "" {

**Review by redhug1 — COMMENTED** [review]
please see the comment questions ...

**Review by redhug1 — APPROVED** [review]
Tests run without any errors in component test log lines.
LGTM

**Pull request merged**